### PR TITLE
Deleting unused `ConfidentialStore.TEST`

### PR DESCRIPTION
**Pull request by jglick**
Cleaning up a package-scoped field that apparently should have been deleted in #4603 but was merely deprecated.

### Testing done

N/A

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7925"></a>

### CODE REVIEW

**Review by NotMyFault — APPROVED** [review]

**Review by timja — APPROVED** [review]

**Comment by NotMyFault** [comment]
/label ready-for-merge

---

This PR is now ready for merge. We will merge it after ~24 hours if there is no negative feedback.
Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process.
Thanks!

**Comment by timja** [comment]
Causes PCT to fail for credentials-plugin: https://ci.jenkins.io/blue/organizations/jenkins/Tools%2Fbom/detail/PR-2051/3/tests/

https://github.com/jenkinsci/credentials-plugin/blob/master/src/test/java/jenkins/security/ConfidentialStoreRule.java

**Review by basil — COMMENTED** [review]
https://github.com/jenkinsci/tfs-plugin/blob/master/tfs/src/test/java/jenkins/security/ConfidentialStoreOverride.java

https://github.com/jenkinsci/kubernetes-cli-plugin/blob/master/src/test/java/jenkins/security/ConfidentialStoreRule.java

**Comment by MarkEWaite** [comment]
I removed the ci.jenkins.io job (sorry about that).  The test results are also stored at https://github.com/jenkinsci/bom/pull/2051/checks?check_run_id=13318131478

**Comment by jglick** [comment]
https://github.com/jenkinsci/tfs-plugin/blob/aea4ce0f1fd82668b1ead7e2aade2ffe36a32a10/pom.xml#L15 is too old.

**Comment by MarkEWaite** [comment]
> https://github.com/jenkinsci/tfs-plugin/blob/aea4ce0f1fd82668b1ead7e2aade2ffe36a32a10/pom.xml#L15 is too old.

I agree that the tfs plugin minimum Jenkins version is badly outdated.  However, that plugin is suspended from distribution due to 3 open security vulnerabilities and no maintainer.  Is the tfs plugin somehow related to this test in a way that I don't understand?

**Comment by jglick** [comment]
Yes, https://github.com/jenkinsci/tfs-plugin/blob/aea4ce0f1fd82668b1ead7e2aade2ffe36a32a10/tfs/src/test/java/jenkins/security/ConfidentialStoreOverride.java#L42 would no longer work in new versions of Jenkins. One of the tasks involved in bringing it up to date, if anyone ever does, would be to delete `ConfidentialStoreOverride` and I guess also `SecretOverride`.

**Pull request merged**
